openedx/features/pakx_features/utils.py

```diff
@@ -0,0 +1,194 @@
+from six import text_type
+from django.db.models.functions import Coalesce
+from django.db.models import Sum, Case, When, IntegerField
+
+from opaque_keys.edx.keys import CourseKey
+from student.models import CourseEnrollment
+from completion.models import BlockCompletion
+from openedx.core.djangoapps.content.block_structure.api import get_course_in_cache
+from openedx.features.course_experience.utils import (get_course_outline_block_tree,
+                                                      get_resume_block)
+from lms.djangoapps.course_api.blocks.serializers import BlockDictSerializer
+from lms.djangoapps.course_api.blocks.transformers.blocks_api import BlocksAPITransformer
+from openedx.core.djangoapps.content.block_structure.transformers import BlockStructureTransformers
+
+
+CORE_BLOCK_TYPES = ['html', 'video', 'problem']
+BLOCK_TYPES_TO_FILTER = ['course', 'chapter', 'sequential', 'vertical', 'discussion', 'openassessment']
+
+
+def add_enrolled_course_progress_and_link(request, course, allow_start_dates_in_future=False):
+    """
+    adds information relevant to resume course functionality to the given course model and progress
+
+    Arguments:
+    :param request: (HttpRequest) request object
+    :param course: (CourseView) course view model object
+    :param allow_start_dates_in_future: (bool) When True, will allow blocks to be
+            returned that can bypass the StartDateTransformer's filter to show
+            blocks with start dates in the future.
+
+    has_visited_course: True if the user has ever visited the course, False otherwise.
+    resume_course_url: The URL of the 'resume course' block if the user has visited the course,
+                        otherwise the URL of the course root.
+    resume_course_title: The display_name of the resume course block, otherwise the display_name of course root
+
+    """
+    course_id = text_type(course.id)
+    course_key = CourseKey.from_string(course_id)
+    course.progress = get_course_progress_percentage(request, course_key)
+    course_id = text_type(course.id)
+    course_outline_root_block = get_course_outline_block_tree(request, course_id, request.user,
+                                                              allow_start_dates_in_future)
+    resume_block = get_resume_block(course_outline_root_block) if course_outline_root_block else None
+    has_visited_course = bool(resume_block)
+    if resume_block:
+        resume_course_url = resume_block['lms_web_url']
+        resume_course_title = resume_block['display_name']
+    else:
+        resume_course_url = course_outline_root_block['lms_web_url'] if course_outline_root_block else None
+        resume_course_title = course_outline_root_block['display_name'] if course_outline_root_block else None
```

Comment:
does this work? 
```
resume course = resume_block if resume_block else course_outline_root_block
resume_course_url = resume_block.get('lms_web_url')
resume_course_title = resume_block.get('display_name')
```

Reply:
The above code is taken from `CourseHomeFragmentView` and we don't know about `course_outline_root_block` value, so there is a double-check for that value, that's why using that exact code here.